rotkehlchen/db/updates.py

```diff
@@ -88,49 +90,111 @@ def update_spam_assets(self, local_version: int, to_version: int) -> None:
         locally
         """
         for version in range(local_version + 1, to_version + 1):
-            file_url = SPAM_ASSETS_URL.format(branch=self.branch, version=version)
-            try:
-                updates = query_file(file_url, is_json=True)
-            except RemoteError as e:
-                log.warning(f'Failed to update spam assets due to {str(e)}')
-                break
-            updated_assets = updates.get('assets')
-            if updated_assets is None:
-                log.error(f'Remote update {file_url} does not contain the assets key')
+            ok, updated_assets = self._get_file_content(
+                file_url=SPAM_ASSETS_URL.format(branch=self.branch, version=version),
+                update_type=UpdateType.SPAM_ASSETS,
+            )
+            if ok is False:
                 continue
 
-            log.info(f'Applying update for spam assets from {version}. {len(updates)} tokens to add')  # noqa: E501
+            log.info(f'Applying update for spam assets from v{version}. {len(updated_assets)} tokens to add')  # noqa: E501
             with GlobalDBHandler().conn.critical_section():
                 # Use a critical section to avoid another greenlet adding spam assets at
                 # the same time
-                update_spam_assets(db=self.user_db, assets_info=updates['assets'])
+                update_spam_assets(db=self.user_db, assets_info=updated_assets)
 
             with self.user_db.conn.write_ctx() as write_cursor:
                 write_cursor.execute(
                     'INSERT OR REPLACE INTO settings(name, value) VALUES (?, ?)',
                     (UpdateType.SPAM_ASSETS.serialize(), version),
                 )
 
+    def update_rpc_nodes(self, local_version: int, to_version: int) -> None:
+        """
+        Check the updates in the inclusive range [local_version + 1, to_version] and update the
+        rpc nodes for those versions.
+        """
+        for version in range(local_version + 1, to_version + 1):
+            ok, updated_rpc_nodes = self._get_file_content(
+                file_url=RPC_NODES_URL.format(branch=self.branch, version=version),
+                update_type=UpdateType.RPC_NODES,
+            )
+            if ok is False:
+                continue
+
+            log.info(f'Applying update for rpc nodes from v{version}')
+            with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                write_cursor.execute('DELETE FROM rpc_nodes')
```

Comment:
explain that this is only the default nodes that change

```suggestion
            with GlobalDBHandler().conn.write_ctx() as write_cursor:
                # Update the default nodes, stores in the global DB
                write_cursor.execute('DELETE FROM rpc_nodes')
```